### PR TITLE
Use yarn pnp as module linker

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
Package manager

### Description
This PR:
- disables the node_modules legacy mode in yarn and enables yarn pnp
- adds some missing peer dependencies
- replaces the cypress config file, which was a typescript file, with a javascript file because otherwise cypress can't read it anymore.
- Removes netlify-cli from the package.json and uses npx in the deploy scripts because the cli manipulates the node_modules directory and therefore doesn't work anymore if installed via yarn.

### Steps

- [x] Added implementation
- [x] Added / updated documentation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/react-client/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### CODE REVIEW

File: .yarnrc.yml [hunk 1]
```diff
@@ -1,5 +1,3 @@
-nodeLinker: node-modules
```
Comment (ErikMichelson) [inline .yarnrc.yml:1]: Maybe set `nodeLinker: pnp` explicitly to ensure consistency with changes in default behaviour in newer yarn versions?

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -66,7 +66,6 @@ jobs:
     needs: build-frontend
     runs-on: ubuntu-latest
     container:
-      # renovate: datasource=docker depName=cypress/browsers versioning=docker
```
Comment (ErikMichelson) [inline .github/workflows/e2e.yml:69]: Why did you remove the renovate update section for the cypress docker image?

Comment (mrdrogdrog) [reply .github/workflows/e2e.yml:69]: It doesn't work

**Review by ErikMichelson — APPROVED** [review]

**Review by DerMolly — APPROVED** [review]
